### PR TITLE
[vlan] Add support of VLAN host interface

### PR DESCRIPTION
**Pull request by volodymyrsamotiy**
* Infrastructure needed for the VNET ping tool

Signed-off-by: Volodymyr Samotiy <volodymyrs@nvidia.com>

**What I did**

 Add support of VLAN host interface

**Why I did it**

It is infrastructure needed for the VNET ping tool

**How I verified it**
Configured new VLAN with "hostif_name" attribute and verified that it created in HW

**Details if related**
N/A

### CODE REVIEW

**Comment by lguohan** [comment]
missing pytest

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -2777,6 +2829,13 @@ void PortsOrch::doVlanTask(Consumer &consumer)
                         gIntfsOrch->setRouterIntfsMac(vl);
                     }
                 }
+                if (!hostif_name.empty())
+                {
+                    if (!createVlanHostIntf(vl, hostif_name))
+                    {
+                        throw runtime_error("Cannot create VLAN host interface");
```
Comment (prsunny) [inline orchagent/portsorch.cpp:2836]: Lets handle the failure gracefully as this is for monitoring Vlan. We can erase` it` and continue.

File: cfgmgr/vlanmgr.cpp [hunk 2]
```diff
@@ -362,6 +363,10 @@ void VlanMgr::doVlanTask(Consumer &consumer)
                     mac = fvValue(i);
                     setHostVlanMac(vlan_id, mac);
                 }
+                else if (fvField(i) == "hostif_name")
```
Comment (prsunny) [inline cfgmgr/vlanmgr.cpp:366]: Does this mean, for every Vlan thats in VNET, a corresponding host if shall be created?

**Comment by volodymyrsamotiy** [comment]
In order for VS test to pass first need to merge the following PR for "vslib": https://github.com/Azure/sonic-sairedis/pull/804

File: cfgmgr/vlanmgr.cpp (hunk 2 above)
Comment (volodymyrsamotiy) [reply cfgmgr/vlanmgr.cpp:366]: Yes, since it is per VLAN attribute and created for specific VLAN.

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (volodymyrsamotiy) [reply orchagent/portsorch.cpp]: Changed.

**Comment by liat-grozovik** [comment]
@prsunny can you please review?

**Comment by prsunny** [comment]
retest this please

**Comment by liat-grozovik** [comment]
> In order for VS test to pass first need to merge the following PR for "vslib": [Azure/sonic-sairedis#804](https://github.com/Azure/sonic-sairedis/pull/804)

@volodymyrsamotiy  it was merged. please check that submodule is updated and then retest.

**Comment by prsunny** [comment]
retest this please

File: orchagent/portsorch.cpp [hunk 3]
```diff
@@ -3689,6 +3753,12 @@ bool PortsOrch::removeVlan(Port vlan)
        return false;
     }
 
+    if (!removeVlanHostIntf(vlan))
```
Comment (prsunny) [inline orchagent/portsorch.cpp:3756]: This must be called only if `host_intf_id ` is present, else skip

File: orchagent/portsorch.cpp [hunk 4]
```diff
@@ -2777,6 +2831,16 @@ void PortsOrch::doVlanTask(Consumer &consumer)
                         gIntfsOrch->setRouterIntfsMac(vl);
                     }
                 }
+                if (!hostif_name.empty())
+                {
+                    if (!createVlanHostIntf(vl, hostif_name))
+                    {
+                        // No need to fail in case of error as this is for monitoring VLAN.
+                        // Error message is printed by "createVlanHostIntf" so just handle faiure gracefully.
```
Comment (prsunny) [inline orchagent/portsorch.cpp:2839]: failure - typo

**Comment by prsunny** [comment]
@volodymyrsamotiy , can you please take a look at the test failures?

File: orchagent/portsorch.cpp (hunk 3 above)
Comment (volodymyrsamotiy) [reply orchagent/portsorch.cpp]: done

File: orchagent/portsorch.cpp (hunk 4 above)
Comment (volodymyrsamotiy) [reply orchagent/portsorch.cpp]: done

**Comment by volodymyrsamotiy** [comment]
Fixed last review comments and test failure

**Review by prsunny — APPROVED** [review]

**Comment by abdosi** [comment]
Need pr for 201911. cherry-pick has conflict.

@prsunny @volodymyrsamotiy

**Comment by abdosi** [comment]
Included without dvs test. Please create new pr for dvs test in 201911.

**Comment by abdosi** [comment]
@volodymyrsamotiy we should optimize the PR to use addHostIntfs() as this can support VLAN based Host router interface. There should not be need of new set of create/remove API's()

@prsunny

**Pull request merged**